wimm.Secundatives/Variant.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+
+namespace wimm.Secundatives
+{
+    /// <summary>
+    /// A typesafe union that contains either a <typeparamref name="T"/> or a <typeparamref name="U"/>
+    /// </summary>
+    public class Variant<T, U>
+    {
+        /// <summary>
+        /// Ensures that <typeparamref name="U"/> and <typeparamref name="T"/> are not within the same
+        /// inheritance heirarchy
+        /// </summary>
+        static Variant()
+        {
+            var uType = typeof(U);
+            var tType = typeof(T);
+
+            if (tType.IsSubclassOf(uType) || uType.IsSubclassOf(tType) )
+            {
+                //TODO:CN -- Maybe fix? Maybe not?
+                throw new NotSupportedException(
+                    "Cannot create a variant of two classess within the same inheritance heirarchy");
+            }
+        }
+
+        //Mike Tyson
+        private readonly object _value;
+
+        /// <summary>
+        /// Checks the type of the type param against the type of the internal member
+        /// </summary>
+        /// <typeparam name="W"> The type to test for </typeparam>
+        /// <returns>Returns true if the internal value is the same as <typeparamref name="W"/></returns>
+        /// <remarks> There is no point calling this method with a value of <typeparamref name="W"/> that 
+        /// is not either <typeparamref name="T"/> or <typeparamref name="U"/> 
+        /// </remarks>
+        public bool Is<W>() => _value is W;
```

Comment:
can the checks not be constrained to T/U? This way a variant<int, string> will have an Is<IPAddress> method.

Reply:
Comment was supposed to read "will have an `Is<IPAddress>` method